### ISSUE TITLE
Coredump Version-4.1.0: zeek::detail::SerializationFormat::ReadData

### ISSUE THREAD

**Issue by initconf** [zeek/zeek#2662]
So I saw this coredump on my 4.1.0 setup - I admit its an older version so feel free to ignore/close this ticket if issue has already been addressed. Putting it here in case if this is something we've not seen before.

Pointer to core file: `~/spool/tmp/post-terminate-manager-2022-12-16-09-20-52-80700-crash/zeek.42386.20201.core`

```
(gdb) bt
#0  0x0000000801a470fd in memcpy () from /lib/libc.so.7
#1  0x000000000086a0c9 in zeek::detail::SerializationFormat::ReadData (this=0x7fffffffe080, b=0x85cf71000, count=5643649511441653571)
    at /usr/local/src/jw/zeek-4.1.0/src/SerializationFormat.cc:75
#2  zeek::detail::BinarySerializationFormat::Read (this=0x7fffffffe080, str=0x8abce0d40, len=0x8abce0d48, tag=<optimized out>)
    at /usr/local/src/jw/zeek-4.1.0/src/SerializationFormat.cc:196
#3  0x00000000008d629f in zeek::threading::Value::Read (this=<optimized out>, fmt=0x7fffffffe080)
    at /usr/local/src/jw/zeek-4.1.0/src/threading/SerialTypes.cc:278
#4  0x0000000000a22eef in zeek::Broker::Manager::ProcessLogWrite (this=<optimized out>, lw=...) at /usr/local/src/jw/zeek-4.1.0/src/broker/Manager.cc:1525
#5  0x0000000000a216e4 in zeek::Broker::Manager::DispatchMessage (this=0x803327180, topic=..., msg=...)
    at /usr/local/src/jw/zeek-4.1.0/src/broker/Manager.cc:1049
#6  0x0000000000a217f3 in zeek::Broker::Manager::DispatchMessage (this=0x803327180, topic=..., msg=...)
    at /usr/local/src/jw/zeek-4.1.0/src/broker/Manager.cc:1068
#7  0x0000000000a23855 in zeek::Broker::Manager::Process (this=0x803327180) at /usr/local/src/jw/zeek-4.1.0/src/broker/Manager.cc:1141
#8  0x00000000008654e0 in zeek::run_state::detail::run_loop () at /usr/local/src/jw/zeek-4.1.0/src/RunState.cc:315
#9  0x00000000006488ce in main (argc=<optimized out>, argv=<optimized out>) at /usr/local/src/jw/zeek-4.1.0/src/main.cc:59
```

**Comment by awelzel** [comment]
> So I saw this coredump on my 4.1.0 setup - I admit its an older version so feel free to ignore/close this ticket if issue has already been addressed. Putting it here in case if this is something we've not seen before.

Three was a known memory corruption that was fixed with 4.2.1 and 4.0.6, but never made it into the 4.1 branch and certainly not fixed in 4.1: https://github.com/zeek/zeek/pull/2040

I'm closing this with the assumption that may have been just that. We'd be very interested in crashes with 5.0.x or 5.1.x :-)

**Comment by awelzel** [comment]
There was also the following known memory corruption around records containing sets being used as keys of tables which also affected 3.0, 4.0 and 4.1.0: https://github.com/zeek/zeek/pull/1772
